tests/e2e/tests/e2e/GitPublishBranch.spec.ts

```diff
@@ -0,0 +1,103 @@
+const projectTree: ProjectTree = e2eContainer.get(CLASSES.ProjectTree);
+const gitPlugin: GitPlugin = e2eContainer.get(CLASSES.GitPlugin);
+const testWorkspaceUtils: TestWorkspaceUtil = e2eContainer.get<TestWorkspaceUtil>(TYPES.WorkspaceUtil);
+
+
+suite('Publish branch in git extension', async () => {
+    const workspacePrefixUrl: string = `${TestConstants.TS_SELENIUM_BASE_URL}/dashboard/#/ide/${TestConstants.TS_SELENIUM_USERNAME}/`;
+    const wsNameGitPublishBranch = WorkspaceNameHandler.generateWorkspaceName('checkGitPublishBranch-', 5);
+    const changedFile = 'README.md';
+    const branchName = WorkspaceNameHandler.generateWorkspaceName('checkGitPublishBranch', 5);
+    const file = 'https://github.com/' + TestConstants.TS_GITHUB_TEST_REPO + '/blob/' + branchName + '/README.md';
+
+    suiteSetup(async function () {
+        const wsConfig = await testWorkspaceUtils.getBaseDevfile();
+        wsConfig.metadata!.name = wsNameGitPublishBranch;
+        await testWorkspaceUtils.createWsFromDevFile(wsConfig);
+    });
+
+    test('Login into workspace', async () => {
+        await driverHelper.navigateToUrl(workspacePrefixUrl + wsNameGitPublishBranch);
+        await loginPage.login();
+        await ide.waitWorkspaceAndIde();
+        await projectTree.openProjectTreeContainer();
+        await driverHelper.wait(15000);
+    });
+
+    test('Create a new branch, change commit and push', async function changeCommitAndPushFunc() {
+        const currentDate: string = Date.now().toString();
+        const readmeFileContentXpath: string = `//div[@id='readme']//p`;
+        await cloneTestRepo();
+
+        await driverHelper.wait(15000);
+        await topMenu.selectOption('View', 'Find Command...');
+        await quickOpenContainer.typeAndSelectSuggestion('branch', 'Git: Create Branch...');
+        await quickOpenContainer.typeAndSelectSuggestion(branchName, `Please provide a new branch name (Press 'Enter' to confirm your input or 'Escape' to cancel)`);
+
+        await projectTree.expandPathAndOpenFile('Spoon-Knife', changedFile);
+        await editor.type(changedFile, currentDate + '\n', 1);
+        await gitPlugin.openGitPluginContainer();
+        await gitPlugin.waitChangedFileInChagesList(changedFile);
+        await gitPlugin.stageAllChanges(changedFile);
+        await gitPlugin.waitChangedFileInChagesList(changedFile);
+        await gitPlugin.typeCommitMessage(this.test!.title + currentDate);
+        await gitPlugin.commitFromCommandMenu();
+        await gitPlugin.pushChangesFromCommandMenu();
+        await driverHelper.waitAndClick(By.xpath(`//button[@class='theia-button main']`));
+        await gitPlugin.waitDataIsSynchronized();
+        await testWorkspaceUtils.cleanUpAllWorkspaces();
+
+        await driverHelper.navigateToUrl(file);
+        const rawDataFromFile: string = await driverHelper.waitAndGetText(By.xpath(readmeFileContentXpath));
```

Comment:
Better to use another approach,
use locator with the expecting text inside and just wait for visibility this locator by using
```
driverHelper.waitVisibility
```

besides maybe better to create method in proper pageobject?